macros/clusterAnaScurve.py

```diff
@@ -0,0 +1,234 @@
+    parser.add_option("--ztrim", type="float", dest="ztrim", default=4.0,
+                      help="Specify the p value of the trim", metavar="ztrim")
+    
+    chanMaskGroup = OptionGroup(
+            parser,
+            "Options for channel mask decisions"
+            "Parameters which specify how Dead, Noisy, and High Pedestal Channels are charaterized")
+    chanMaskGroup.add_option("--maxEffPedPercent", type="float", dest="maxEffPedPercent", default=0.05,
+                      help="Percentage, Threshold for setting the HighEffPed mask reason, if channel (effPed > maxEffPedPercent * nevts) then HighEffPed is set",
+                      metavar="maxEffPedPercent")
+    chanMaskGroup.add_option("--highNoiseCut", type="float", dest="highNoiseCut", default=1.0,
+                      help="Threshold for setting the HighNoise maskReason, if channel (scurve_sigma > highNoiseCut) then HighNoise is set",
+                      metavar="highNoiseCut")
+    chanMaskGroup.add_option("--deadChanCutLow", type="float", dest="deadChanCutLow", default=4.14E-02,
+                      help="If channel (deadChanCutLow < scurve_sigma < deadChanCutHigh) then DeadChannel is set",
+                      metavar="deadChanCutLow")
+    chanMaskGroup.add_option("--deadChanCutHigh", type="float", dest="deadChanCutHigh", default=1.09E-01,
+                      help="If channel (deadChanCutHigh < scurve_sigma < deadChanCutHigh) then DeadChannel is set",
+                      metavar="deadChanCutHigh")
+    parser.add_option_group(chanMaskGroup)
+    
+    (options, args) = parser.parse_args()
+    listOfScanDatesFile = options.filename
+
+    # Check if the queue is supported
+    # See: https://cern.service-now.com/service-portal/article.do?n=KB0000470
+    import os
+    from anaInfo import queueNames, tree_names
+    if options.queue not in queueNames:
+        print("queue '%s' not understood"%options.queue)
+        print("list of supported queues is:", queueNames)
+        exit(os.EX_USAGE)
+        pass
+
+    # Check anaType is understood
+    supportedAnaTypes = ['scurve','trim']
+    if options.anaType not in supportedAnaTypes:
+        print("Invalid analysis specificed, please select only from the list:")
+        print(supportedAnaTypes)
+        exit(os.EX_USAGE)
+        pass
+
+    # Prepare the commands for making the
+    from gempython.utils.wrappers import envCheck, runCommand
+    envCheck('BUILD_HOME')
+    envCheck('DATA_PATH')
+    envCheck('ELOG_PATH')
+
+    buildHome= os.getenv('BUILD_HOME')
+    dataPath = os.getenv('DATA_PATH')
+    elogPath = os.getenv('ELOG_PATH')
+
+    # Get info from input file
+    from anautilities import getDirByAnaType, filePathExists, makeListOfScanDatesFile, parseListOfScanDatesFile
+    if (listOfScanDatesFile is None and options.chamberName is not None):
+        makeListOfScanDatesFile(options.chamberName, options.anaType, ztrim=options.ztrim)
```

Comment:
I would add the possibility to set start/end scan dates here. I.e. I have no list of scan dates file (or have a wrong one), but I want to have one automatically created (updated) for certain desired dates. In a current implementation the new list will be created for all dates from the oldest one till today

Reply:
Yes this can be implemented.